.github/workflows/unittest.yml

```diff
@@ -0,0 +1,114 @@
+name: UnitTests
+
+on: [push, pull_request]
+
+jobs:
+  ubuntu:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        use-conda: [true, false]
+        context: ['fork', 'spawn', 'forkserver']
+        use-dist: [false]
+        scikit-learn: [0.22.2, 0.23.1]
+        include:
+          - python-version: 3.8
+            code-cov: true
+            scikit-learn: 0.22.2
+            context: 'fork'
+            use-dist: false
+          - python-version: 3.7
+            scikit-learn: 0.22.2
+            use-conda: false
+            use-dist: true
+            context: 'fork'
+          - python-version: 3.6
+            use-conda: false
+            scikit-learn: 0.19.2
+            context: 'fork'
+            use-dist: false
+        exclude:
+          # Older python cannot move threadlock objects around
+          - python-version: 3.6
+            context: 'spawn'
+          - python-version: 3.6
+            context: 'forkserver'
+      fail-fast:  false
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Conda Install test dependencies 
+      if: matrix.use-conda == true
+      run: |
+        # Miniconda is available in $CONDA env var
+        $CONDA/bin/conda create -n testenv --yes pip wheel gxx_linux-64 gcc_linux-64 swig python=${{ matrix.python-version }}
+        $CONDA/envs/testenv/bin/python3 -m pip install --upgrade pip
+        $CONDA/envs/testenv/bin/pip3 install -e .[test]
+    - name: Install test dependencies 
+      if: matrix.use-conda == false && matrix.use-dist == false
+      run: |
+        python -m pip install --upgrade pip
+        pip install -e .[test]
+        sudo apt-get update
+    - name: Dist Install test dependencies 
+      if: matrix.use-conda == false && matrix.use-dist == true
+      run: |
+        python -m pip install --upgrade pip
+        sudo apt-get update
+        # We need to install for the dependencies, like pytest
+        pip install -e .[test]
+        pip uninstall --yes pynisher
+        python setup.py sdist
+        last_dist=$(ls -t dist/pynisher-*.tar.gz | head -n 1)
+        pip install $last_dist
```

Comment:
Would `pip install ${last_dist}[test] work and simplify this script a bit?

Reply:
thanks for the suggestion. We should do the same for autosklearn then.